app_flutter/lib/commit_box.dart

```diff
@@ -45,67 +45,70 @@ class _CommitBoxState extends State<CommitBox> {
   }
 
   void _handleTap() {
-    _commitOverlay = this._createCommitOverlay(widget);
+    _commitOverlay = CommitOverlay(widget, context);
     Overlay.of(context).insert(this._commitOverlay);
   }
+}
 
-  OverlayEntry _createCommitOverlay(CommitBox widget) {
-    RenderBox renderBox = context.findRenderObject();
+class CommitOverlay extends OverlayEntry {
```

Comment:
Maybe something like this:

```
class CommitOverlayContents extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Stack(...);
  }
}

...

Overlay.of(context).insert(OverlayEntry(builder: (buildContext context) => CommitOverlayContents());

```

Reply:
You could pass in the callback to remove in the constructor to `CommitOverlayContents`

Reply:
Sweet! Thank you, that was the last piece CommitOverlayContents needed :)